src/Symfony/Component/HttpKernel/EventListener/ExceptionListener.php

```diff
@@ -62,7 +62,7 @@ public function onKernelException(GetResponseForExceptionEvent $event)
         } catch (\Exception $e) {
             $f = FlattenException::createFromThrowable($e);
 
-            $this->logException($e, sprintf('Exception thrown when handling an exception (%s: %s at %s line %s)', $f->getClass(), $f->getMessage(), $e->getFile(), $e->getLine()));
+            $this->logThrowable($e, sprintf('Exception thrown when handling an exception (%s: %s at %s line %s)', $f->getClass(), $f->getMessage(), $e->getFile(), $e->getLine()));
```

Comment:
I don't see the value of this as you only `} catch (\Exception $e) {`. So `$e` is always an `Exception` currently. Also this break for custom listeners that only overwrite `logException` which is not called anymore.

What I had planned is that we do not create a sub-request anymore for exceptions, but call the ErrorRenderer redirectly. This would probably mean, we deprecate the whole ExceptionListener and create a new one. This would also solve the `duplicateRequest` problem you mentioned. So it might not be worth to worry about ExceptionListener if we replace it completely anyway.

Reply:
`logThrowable()` is needed by `logKernelException()`. Since `logException()` becomes deprecated, I just replaced its calls with the new method.

> What I had planned is that we do not create a sub-request anymore for exceptions, but call the ErrorRenderer redirectly. This would probably mean, we deprecate the whole ExceptionListener and create a new one. This would also solve the duplicateRequest problem you mentioned. So it might not be worth to worry about ExceptionListener if we replace it completely anyway.

👌 👍

Reply:
> What I had planned is that we do not create a sub-request anymore for exceptions, but call the ErrorRenderer redirectly. This would probably mean, we deprecate the whole ExceptionListener and create a new one. This would also solve the duplicateRequest problem you mentioned. So it might not be worth to worry about ExceptionListener if we replace it completely anyway.

Maybe I got it wrong, but the sub-request aims to call a custom exception controller, remove it and use error rendering directly would also mean removing the exception controller layer/feature, which is currently the easiest way to plug in to system exceptions.